### ISSUE TITLE
Bug: Cannot use typed variables (More specifically those from Structs) in loops

### ISSUE THREAD

**Issue by pielover928** [ItsDeltin/Overwatch-Script-To-Workshop#59]

**Comment by ItsDeltin** [comment]
Do you have the code that causes this?

**Comment by pielover928** [comment]
Define globalvar markers = [];
Append(markers, new Marker);

Foreach(Marker marker in markers){
   Marker.pos = something;
}

This doesn't create an error in the code itself but rather puts a red underline on the rule's event line.

It also happens when I make marker a generic variable and cast to Marker. I think that it's just a side effect of not being able to have typed arrays.

**Comment by ItsDeltin** [comment]
Workaround:
```
foreach (define a in structArray)
    {
        myStruct b = a;
    }
```

**Comment by ItsDeltin** [comment]
Fixed in fe5889100212a30754eda75bc8de2f9a01e3cbbb

**Comment by ItsDeltin** [comment]
Fixed in [v0.5.](https://github.com/ItsDeltin/Overwatch-Script-To-Workshop/releases/tag/v0.5)
